### PR TITLE
log-backup: directly override the exists file when do truncate on gcs or azure

### LINKED ISSUE #38229: log-backup: failed to truncate the log data on gcs or azure blob
## Bug Report

Please answer these questions before submitting your issue. Thanks!

### 1. Minimal reproduce step (Required)
do log truncate

### 2. What did you expect to see? (Required)
truncate done.
### 3. What did you see instead (Required)
Error: failed during rewriting the file at v1_stream_trancate_safepoint.txt in azure://test/pitr-test-01: the origin file doesn't exist: [BR:Common:ErrInvalidArgument]invalid argument
### 4. What is your TiDB version? (Required)
nightly

### PR DESCRIPTION
**Pull request by Leavrth**
### What problem does this PR solve?

Issue Number: close #38229 

Problem Summary:
the file on gcs or azure can be directly override when there is another file uploaded with the same name.
### What is changed and how it works?
override the exists file when do truncate on gcs or azure.
### Check List

Tests 

- [ ] Unit test
- [ ] Integration test
- [x] Manual test (add detailed scripts or steps below)
- [ ] No code

### Release note

Please refer to [Release Notes Language Style Guide](https://pingcap.github.io/tidb-dev-guide/contribute-to-tidb/release-notes-style-guide.html) to write a quality release note.

```release-note
None
```

### CODE REVIEW

**Review by 3pointer — APPROVED** [review]

File: br/pkg/restore/stream_metas_test.go [hunk 1]
```diff
@@ -246,6 +247,46 @@ func TestTruncateSafepoint(t *testing.T) {
 	}
 }
 
+func TestTruncateSafepointForGCS(t *testing.T) {
```
Comment (joccau) [inline br/pkg/restore/stream_metas_test.go:250]: do we need to test azblob storage ?

**Review by joccau — COMMENTED** [review]
rest LGTM

File: br/pkg/restore/stream_metas_test.go (hunk 1 above)
Comment (Leavrth) [reply br/pkg/restore/stream_metas_test.go:250]: Currently, the unit test and integration test for azure are not supported.

**Review by joccau — APPROVED** [review]
LGTM

**Review by YuJuncen — APPROVED** [review]

**Pull request merged**